pkg/types/types.go

```diff
@@ -53,6 +54,9 @@ func Init(nodeName string) {
 }
 
 type CommonData struct {
+	// Timestamp
+	Timestamp string `json:"timestamp,omitempty" column:"timestamp,template:timestamp"`
```

Comment:
I'm wondering about the format to use here. Isn't it better to use an integer (uint64) and let the consumers do the formatting?

Reply:
I pushed a second patch using an integer.

@flyth But I am not sure how to use `MustSetExtractor` in a generic way to display the date without duplicating the code for every gadget. Is there a way to define an extractor in the template?

Reply:
A pretty generic solution could be to make the columns lib aware of the `fmt.Stringer` interface:

```golang
type Stringer interface {
	String() string
}
```

And then let the Timestamp field have a type Int64 that implements that interface:
```golang
func (v Int64) String() string {
	return time.Format(...)
}
```

[Here's](https://github.com/inspektor-gadget/inspektor-gadget/commit/c9c3c705befbdc62d8e2c6c946eb78e4244cf84b) an implementation of that for the lib - if you think this is a good way (although not 100% optimized, yet), let me know and I'll create a PR.

In that case you would just have to add `,stringer` to the column definition.

Reply:
@flyth That sounds good. But I didn't understand in which package would the function `func (v Int64) String() string` be implemented.

Reply:
I'd suggest in [pkg/types/types.go](https://github.com/inspektor-gadget/inspektor-gadget/pull/1205/files#diff-8e49337a1417a39a62ebf427e84fb2faa566c96e37ae627ac484c05dbe65c483), something like this:

```golang
type Time int64

func (t Time) String() string {
	return time.Format(...)
}

type CommonData struct {
 	// Timestamp in nanoseconds since January 1, 1970 UTC. A int64 is big
 	// enough to represent time between the year 1678 and 2262.
 	Timestamp Time `json:"timestamp,omitempty" column:"timestamp,template:timestamp,stringer"`
}
```

Reply:
I updated this PR with that approach. I picked up c9c3c705befb and added another patch to fix a panic in the column package.